electrum/transport_ws.go

```diff
@@ -0,0 +1,104 @@
+package electrum
+
+import (
+	"context"
+	"crypto/tls"
+	"log"
+	"time"
+
+	"github.com/gorilla/websocket"
+)
+
+type WebSocketTransport struct {
+	conn      *websocket.Conn
+	responses chan []byte
+	errors    chan error
+}
+
+// NewWebSocketTransport initializes new WebSocket transport.
+func NewWebSocketTransport(
+	ctx context.Context,
+	url string,
+	tlsConfig *tls.Config,
+) (*WebSocketTransport, error) {
+	dialer := websocket.Dialer{
+		TLSClientConfig: tlsConfig,
```

Comment:
Should we enforce that `tlsConfig` must be `nil` for `ws` connections or that's fine to not bother about it at all? Conversely, shouldn't we enforce it is non-nil for `wss`?